server/game/gameSystems/DiscardEntireHandSystem.ts

```diff
@@ -0,0 +1,44 @@
+import { EventName, Location } from '../core/Constants';
+import { AbilityContext } from '../core/ability/AbilityContext';
+import * as Helpers from '../core/utils/Helpers.js';
+import type Player from '../core/Player';
+import { IPlayerTargetSystemProperties, PlayerTargetSystem } from '../core/gameSystem/PlayerTargetSystem';
+import { DiscardSpecificCardSystem } from './DiscardSpecificCardSystem';
+
+export interface IDiscardEntireHandSystemProperties extends IPlayerTargetSystemProperties {
+    target: Player | Player[];
```

Comment:
Oh interesting, I didn't realize this wasn't being explicitly overridden in IPlayerTargetSystemProperties. Do you mind changing it there to be:

```typescript
export interface IPlayerTargetSystemProperties extends IGameSystemProperties {
    target?: Player | Player[];
}
```

Then this can just be an empty class (we always define one even if empty just for consistency)

Reply:
I'll add it to `IPlayerTargetSystemProperties` too but I would keep it here too without the optionality to force you to pass a target. I think the default behaviour of discarding the opponent hand is a bit counterintuitive and it'd be better to always specify the target.

Reply:
I agree but unfortunately the pattern we use with target resolvers requires that the target property is always optional. Since target resolvers will populate the target field at resolution time, we don't provide it when we initialize the system. So right now this system couldn't be used inside a PlayerTargetResolver

Reply:
Wouldn't you be able to do:
```Typescript
AbilityHelper.immediateEffects.discardEntireHand((context) => ({ target: context.target }))
```
when using `PlayerTargetResolver`?